File: around/feature/backend/app/models/tour.py
from pydantic import BaseModel, Field


class Tour(BaseModel):
    id: int
    business_user_id: int
    title: str
    description: str
    duration_days: int
    price: float
    distance_km: float
    stops_count: int
    difficulty: str
    is_published: bool


class TourCreateIn(BaseModel):
    title: str = Field(min_length=3, max_length=120)
    description: str = Field(min_length=3, max_length=2000)
    duration_days: int = Field(ge=1, le=60)
    price: float = Field(ge=0, le=1_000_000)
    distance_km: float = Field(gt=0, le=5000)
    stops_count: int = Field(default=1, ge=1, le=200)
    difficulty: str = Field(default="easy", pattern="^(easy|medium|hard)$")
    is_published: bool = False


class TourUpdateIn(BaseModel):
    title: str | None = Field(default=None, min_length=3, max_length=120)
    description: str | None = Field(default=None, min_length=3, max_length=2000)
    duration_days: int | None = Field(default=None, ge=1, le=60)
    price: float | None = Field(default=None, ge=0, le=1_000_000)
    distance_km: float | None = Field(default=None, gt=0, le=5000)
    stops_count: int | None = Field(default=None, ge=1, le=200)
    difficulty: str | None = Field(default=None, pattern="^(easy|medium|hard)$")
    is_published: bool | None = None

File: around/feature/frontend/lib/models/tour.dart
class Tour {
  final int id;
  final int businessUserId;
  final String title;
  final String description;
  final int durationDays;
  final double price;
  final double distanceKm;
  final int stopsCount;
  final String difficulty;
  final bool isPublished;

  Tour({
    required this.id,
    required this.businessUserId,
    required this.title,
    required this.description,
    required this.durationDays,
    required this.price,
    required this.distanceKm,
    required this.stopsCount,
    required this.difficulty,
    required this.isPublished,
  });

  factory Tour.fromJson(Map<String, dynamic> json) {
    return Tour(
      id: json['id'] as int,
      businessUserId: json['business_user_id'] as int,
      title: json['title'] as String,
      description: json['description'] as String,
      durationDays: json['duration_days'] as int,
      price: (json['price'] as num).toDouble(),
      distanceKm: (json['distance_km'] as num).toDouble(),
      stopsCount: json['stops_count'] as int,
      difficulty: json['difficulty'] as String,
      isPublished: json['is_published'] as bool,
    );
  }
}
